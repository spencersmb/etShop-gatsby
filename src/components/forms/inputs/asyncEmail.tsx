import { CheckoutApi } from '@api/checkoutApi'
import { ICouponApiResponse } from '@et/types/Cart'
import { InputError, SvgValidation } from '@styles/modules/SignInUpModals'
import { svgs } from '@svg'
import { useSetState } from '@utils/stateUtils'
import React, { useEffect, useRef, useState } from 'react'
import { from, fromEvent, Observable } from 'rxjs'
import { debounceTime, distinctUntilChanged, filter, map, switchMap } from 'rxjs/operators'
import { Subscription } from 'rxjs/src/internal/Subscription'
// import {colors} from '@et/styles/base/colors'
import styled from 'styled-components'
import { renderSvg } from '@utils/styleUtils'

interface InputField {
	name: string,
	onBlur: () => void,
	onChange: (el: any) => void,
	onDragStart: () => void,
	onDrop: () => void,
	onFocus: () => void,
	value: any
}

export interface InputMeta {
	active: boolean,
	asyncValidating: boolean,
	autofilled: boolean,
	dirty: boolean,
	dispatch: () => void,
	error: any,
	form: string,
	initial: any,
	invalid: boolean,
	pristine: boolean,
	submitFailed: boolean,
	submitting: boolean,
	touched: boolean,
	valid: boolean,
	warning: any
}

interface IProps {
	input: InputField,
	label: string,
	meta: InputMeta,
	type: string,
	className?: string,
	value?: any,
	id?: string,
	name?: string,
	key?: string,
	placeholder?: string,
	disabled?: boolean,
	svg?: string,
	price?: string,
	autoComplete?: boolean,
	setEmailTaken: (state: boolean) => void
	emailTaken: boolean
}

export const RxEmailField = (props: IProps) => {
	const [state, setState] = useSetState({
		loading: false,
		response: false
	})
	console.log('props async', props)

	const { loading, response } = state
	const inputRef = useRef<HTMLInputElement | null>(null)
	const {
		key,
		input,
		label,
		type,
		disabled,
		placeholder,
		svg,
		setEmailTaken,
		emailTaken,
		meta: { asyncValidating, pristine, touched, invalid, active, dirty, error, warning }
	} = props

	const renderSvgColor: string = !pristine || touched
		? invalid || emailTaken
			? 'red'
			: 'green'
		: 'grey' // silver

	// console.log('error', error)
	// console.log('pristine', pristine)
	// console.log('invalid', invalid)
	// console.log('touched', touched)

	const showLabel = (): boolean => {
		if (active) {
			return true
		} else if ((dirty && !pristine) || input.value) {
			return true
		}
		return false
	}

	const messageTest = (errorObj: {} | string) => {
		if (typeof errorObj === 'string') {
			return (
				<span>{error}</span>
			)
		}
	}

	const showIcon = () => {

		return !pristine || touched
			? (invalid)
				? <SvgValidation isValid={false}>
					{renderSvg(svgs.Close)}
				</SvgValidation>
				: <SvgValidation isValid={true}>
					{renderSvg(svgs.Checkmark)}
				</SvgValidation>
			: null
	}

	const isFocused = active ? 'hasFocus' : 'noFocus'
	const isValid = !invalid ? 'valid' : 'invalid'
	const isEmpty = input.value === '' ? 'empty' : 'has-value'

	return (
		<>
			<div key={key} className={
				`formGroup ${isFocused} ${isValid} ${isEmpty}`
			}>
				<label
					htmlFor={input.name}
					className={'renderLabel'}
				>{label}</label>
				<input
					id={input.name}
					ref={inputRef}
					{...input}
					aria-label={label}
					placeholder={placeholder}
					type={type}
					disabled={disabled}
					readOnly={disabled}
				/>
				<div style={{ position: 'absolute', right: 0, top: 0 }}>{JSON.stringify(asyncValidating)}</div>

				{!loading && showIcon()}
				{/*{messageTest(warning)}*/}
			</div>
			{!invalid && emailTaken && <InputError>
        <span>Sorry, email is already in use</span>
      </InputError>}
			{touched && <InputError>{messageTest(error)}</InputError>}
		</>
	)
}

export default RxEmailField

export const Svg = styled.div`
	path, rect{
		fill: ${(props: any) => props.color};
	}
	width: 25px;
`

interface ILabelProps {
	active: boolean;
	spencer?: string;
}

// Optional Styled-Components
// const Label = styled<ILabelProps, 'label'>('label')`
// 	display: block;
// 	color: ${(props) => props.active ? 'red' : 'blue'};
// `
// had to export this to test
export const Label = styled.label`
	display: block;
	color: ${(props: ILabelProps) => props.active ? 'green' : 'blue'};
`
