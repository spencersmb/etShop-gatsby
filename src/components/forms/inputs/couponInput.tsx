import { ICouponState } from '@et/types/Cart'
import { IState } from '@et/types/State'
import { submitCouponCode } from '@redux/actions/couponActions'
import React, { useEffect, useState } from 'react'
import { connect } from 'react-redux'
import { Action, bindActionCreators, Dispatch } from 'redux'

interface IReduxActions {
	checkCoupon: (couponCode: string) => void
}

interface IProps {
	coupon: ICouponState
}

export function CouponInput (props: IProps & IReduxActions) {
	const [input, setInput] = useState('')
	const { coupon, checkCoupon } = props

	useEffect(() => {
		if (coupon.valid) {
			setInput(coupon.code)
		}
	}, [])

	function handleSubmit (e: any) {
		e.preventDefault()
		if (input !== '') {
			checkCoupon(input)
		}
	}

	function handleInputChange (e: any) {
		setInput(e.target.value)
	}

	return (
		<form onSubmit={handleSubmit}>
			{coupon.valid && <span data-testid='valid-notice'>Valid code!</span>}
			{!coupon.valid && coupon.submitted && <span data-testid='invalid-notice'>Invalid code!</span>}
			<label htmlFor='coupon'>Coupon</label>
			<input
				aria-label={'coupon'}
				type='text'
				onChange={handleInputChange}
				data-testid='couponInput'
				value={input}
				readOnly={coupon.loading}/>
			<button type='submit' data-testid='couponSubmitBtn'
							disabled={coupon.loading || input === ''}>{coupon.loading ? 'Checking...' : 'Submit'}</button>
		</form>
	)
}

const mapStateToProps = (state: IState): { coupon: ICouponState } => {
	return {
		coupon: state.cart.coupon
	}
}
const mapDispatchToProps = (dispatch: Dispatch<Action>) => {
	return {
		checkCoupon: bindActionCreators(submitCouponCode, dispatch)
	}
}
export default connect<IProps, IReduxActions, {}, IState>(mapStateToProps, mapDispatchToProps)(CouponInput)