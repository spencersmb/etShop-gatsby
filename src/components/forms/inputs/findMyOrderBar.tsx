import AuthApi from '@api/authApi'
import SubmitButton from '@components/buttons/submitButton'
import { device } from '@styles/global/breakpoints'
import { colors } from '@styles/global/colors'
import { Sentinel } from '@styles/global/fonts'
import { placeholderColor } from '@styles/global/mixins'
import { shadowStyles } from '@styles/global/shadows'
import { svgs } from '@svg'
import { fakeApiCall } from '@utils/apiUtils'
import { useSetState } from '@utils/stateUtils'
import { renderSvg } from '@utils/styleUtils'
import React from 'react'
import posed from 'react-pose'
import styled from 'styled-components'

interface IProps {
	handleSubmit: any
	handleInputChange: any
	state: any
}

const FindMyOrderBar = (props: IProps) => {
	const { state } = props

	const handleSubmit = async (e: any) => {
		e.preventDefault()
		props.handleSubmit()
	}

	return (
		<FindOrderWrapper>
			<form onSubmit={handleSubmit}>
				<div className='formInner'>
					<div className={`inputWrapper`}>
						<span>
							{renderSvg(svgs.Tag)}
						</span>
						<input
							required={true}
							type='text'
							value={state.orderId}
							placeholder={`Order number`}
							onChange={props.handleInputChange}
							data-input={`orderId`}/>
					</div>
					<div className={`inputWrapper`}>
						<span>
							{renderSvg(svgs.Email)}
						</span>
						<input
							required={true}
							type='email'
							value={state.email}
							placeholder={`Email Address`}
							onChange={props.handleInputChange}
							data-input={`email`}/>
					</div>
				</div>
				<div className='buttonWrapper'>
					<SubmitButton
						textColor={'#fff'}
						buttonText={'Search'}
						backgroundColor={colors.db.primary}
						spinnerColor={colors.teal.i500}
						spinnerBgColor={colors.db.primary}
						submitting={state.submitting}
						invalid={state.submitting}
						theme={{
							submitButton: {
								size: 'large',
								font: `${Sentinel.semiboldItalic}`
							}
						}}
					/>
				</div>
			</form>
		</FindOrderWrapper>
	)
}

export default FindMyOrderBar

const FindOrderWrapper = styled.div`
	margin-top: 30px;
	margin-bottom: 30px;
	position: relative;
	z-index: 3;

	form{
		display: flex;
		flex-direction: column;
		@media ${device.tablet} {
			box-shadow: ${shadowStyles.shadow6};
			flex-direction: row;
			border-radius: 50px;
			overflow: hidden;
			background: #fff;
			padding: 0 10px;
		}
		
		@media ${device.laptop} {
			padding-left: 15px;
		}
			
	}
	.formInner{
		display: flex;
		flex-direction: column;
		background: #fff;
		border-radius: 15px;
		overflow: hidden;
		box-shadow: ${shadowStyles.shadow6};
		
		@media ${device.tablet} {
			flex-direction: row;
			box-shadow: none;
			border-radius: 0;
			flex: 1;
		}
	}
	
	.buttonWrapper{
		margin-top: 20px;
		
		@media ${device.tablet} {
			margin-top: 0;
			padding: 10px 0;
		}
	}
	
	.inputWrapper{
		position: relative;
		padding-left: 35px;
		flex: 1;
		span{
			top: 50%;
			left: 12px;
			transform: translateY(-50%);
			position: absolute;
			width:30px;
			height:30px;
			svg{
				width: 100%;
			}		
			path{
				fill: ${colors.purple.i500};
			}
		}
		&:first-child{
			border-bottom: 1px solid #ddd;
		}
		
		@media ${device.tablet} {
			display: flex;
			&:first-child{
				border-bottom: none;
				flex: .7;
				border-right: 1px solid #ddd;
			}	
		    
		}
		
		@media ${device.laptop} {
			margin-left: 10px;
			&:first-child{
				flex: .5;
			}	
		}
			
			
	}
	
	input{
		border: none;
		margin: 0;
		width: 100%;
		padding: 20px 15px;
		${placeholderColor(colors.secondary.text)};
		${Sentinel.semiboldItalic};
		color: ${colors.primary.headline};
		&:focus{
			outline: none;
		}
		
		@media ${device.tablet} {
			padding: 15px;
		}
	}
`
