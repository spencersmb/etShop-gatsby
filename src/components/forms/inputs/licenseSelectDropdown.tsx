import React, { ChangeEvent, ReactEventHandler } from 'react'

interface IProps {
	hasExtendedLicesnse: boolean
	change: any
	selected: string
}

const LicenseSelectDropdown = (props: IProps) => {
	const { hasExtendedLicesnse, change, selected } = props

	function handleChange (e: ChangeEvent<HTMLSelectElement>) {
		change(e.target.value)
	}

	if (hasExtendedLicesnse) {
		return (
			<select onChange={handleChange} value={selected} data-testid={'license_select'}>
				<option value='standard'>Standard License</option>
				<option value='extended'>Extended License</option>
			</select>
		)
	} else {
		return (
			<select>
				<option value='standard'>Standard License</option>
			</select>
		)
	}
}

export default LicenseSelectDropdown
