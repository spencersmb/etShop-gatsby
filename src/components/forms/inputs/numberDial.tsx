import React, { ChangeEvent, useEffect, useState } from 'react'

/**
 * NumberDial properties.
 */
export interface IProps {

	/** prop1 description */
	qty: number | string;
	disableInput?: boolean;
	inputOnChange: (total: number | string) => void
	label?: string
	className?: string
}

function NumberDial ({ qty = 0, inputOnChange, disableInput, className, label = '' }: IProps) {
	const handleInputOnChange = (e: ChangeEvent<HTMLInputElement>) => {
		// Match OBJECT for unit testing purposes
		if (e.target instanceof HTMLInputElement || typeof e.target === 'object'
		) {
			const inputValue: string = e.target.value

			let currentNumber: number | string = 0

			if (inputValue.length > 3) {
				return
			}

			if (inputValue) {
				currentNumber = parseInt(inputValue, 10)
			}

			if (inputValue === '') {
				currentNumber = ''
			}

			inputOnChange(currentNumber)
		}
	}

	return (
		<div className={className}>
			<label htmlFor='numberDial'>{label}</label>
			<input
				data-testid='numberDial'
				id='numberDial'
				type='number'
				className='numberInput'
				onChange={handleInputOnChange}
				value={qty}
				onKeyPress={(e) => {
					if (e.key === '.' || e.key === '-') {
						return
					}
				}}
				readOnly={disableInput}
			/>
		</div>
	)
}

export default NumberDial
