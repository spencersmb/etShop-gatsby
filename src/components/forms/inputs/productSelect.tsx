import LicenseCard from '@components/cards/licenseCard'
import { IProductBullet, IProductLicenseType } from '@et/types/Products'
import { IShowModalAction } from '@redux/actions/modalActions'
import { calcBulkPriceDiscount } from '@utils/priceUtils'
import React, { SyntheticEvent } from 'react'
import styled from 'styled-components'

interface IProps {
	standardLicPrice: string,
	extendedLicPrice?: string,
	license: IProductLicenseType
	onChange: any,
	selectedLicense: string,
	inCart?: boolean
	showModal: IShowModalAction,
	bulkDiscount: boolean
	licenceQty: number
}

const ProductSelect = (
	{ standardLicPrice, extendedLicPrice, onChange, selectedLicense, inCart, license, showModal, bulkDiscount, licenceQty }: IProps) => {

	// console.log('license', license)

	function handleLicClick (e: SyntheticEvent) {
		e.preventDefault()

		if (e.currentTarget) {
			onChange(e.currentTarget.getAttribute('data-lic'))
		}

	}

	function triggerViewLicense (e: any) {
		e.preventDefault()
		showModal({
			modal: () => (<div>License</div>),
			options: {
				closeModal: true,
				hasBackground: true,
				data: {
					test: 'spencer'
				}
			}
		})
	}

	if (license.hasExtendedLicense) {
		return (
			<LicenseCardWrapper data-testid='wrapper'>

				{/*standard lic*/}
				<LicenseCard
					inCart={inCart || false}
					isSelected={selectedLicense === 'standard'}
					price={calcBulkPriceDiscount(bulkDiscount, standardLicPrice, licenceQty)}
					type='standard'
					title='Standard License'
					desc={license.standardItem.desc}
					// bullets={license.standardItem.bullets}
					handleViewLicense={triggerViewLicense}
					handleLicenseClick={handleLicClick}
				/>

				{/*extended lic*/}
				{extendedLicPrice &&
        <LicenseCard
          inCart={inCart || false}
          isSelected={selectedLicense === 'extended'}
          price={calcBulkPriceDiscount(bulkDiscount, extendedLicPrice, licenceQty)}
          type='extended'
          title='Extended License'
          desc={license.standardItem.desc}
					// bullets={license.extendedItem.bullets}
          handleViewLicense={triggerViewLicense}
          handleLicenseClick={handleLicClick}
        />}

			</LicenseCardWrapper>
		)
	}

	return (
		<LicenseCardWrapper data-testid='wrapper'>
			{/*standard lic*/}
			<LicenseCard
				inCart={inCart || false}
				isSelected={selectedLicense === 'standard'}
				price={calcBulkPriceDiscount(bulkDiscount, standardLicPrice, licenceQty)}
				type='standard'
				title='Standard License'
				desc={license.standardItem.desc}
				handleViewLicense={triggerViewLicense}
				handleLicenseClick={handleLicClick}
			/>
		</LicenseCardWrapper>
	)

}

const LicenseCardWrapper = styled.div`
	margin-bottom: 15px;
	width: 100%;
`

export default ProductSelect
