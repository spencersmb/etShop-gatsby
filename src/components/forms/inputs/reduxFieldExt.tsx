import React, { Component } from 'react'
import { Field, Validator } from 'redux-form'

interface IReduxField {
	name: string;
	type: string;
	component: any;
	placeholder: string;
	label: string;
	disabled?: boolean;
	svg?: any;
	validate?: Validator | Validator[];
	warn?: Validator | Validator[];
	withRef?: boolean;
	// dirty: boolean;
	// pristine: boolean;
	// value: any;
}

export class ReduxFieldExt extends React.Component<IReduxField>{
	constructor(props: IReduxField) {
		super(props)
	}
	render(){
		return(
			<>
				<Field {...this.props}/>
			</>
		)
	}
}

export default ReduxFieldExt