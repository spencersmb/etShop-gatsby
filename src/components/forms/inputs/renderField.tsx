import React from 'react'
// import {colors} from '@et/styles/base/colors'
import styled from 'styled-components'
import { renderSvg } from '@utils/styleUtils'

interface InputField {
	name: string,
	onBlur: () => void,
	onChange: (el: any) => void,
	onDragStart: () => void,
	onDrop: () => void,
	onFocus: () => void,
	value: any
}

export interface InputMeta {
	active: boolean,
	asyncValidating: boolean,
	autofilled: boolean,
	dirty: boolean,
	dispatch: () => void,
	error: any,
	form: string,
	initial: any,
	invalid: boolean,
	pristine: boolean,
	submitFailed: boolean,
	submitting: boolean,
	touched: boolean,
	valid: boolean,
	warning: any
}

interface IProps {
	input: InputField,
	label: string,
	meta: InputMeta,
	type: string,
	className?: string,
	value?: any,
	id?: string,
	name?: string,
	key?: string,
	placeholder?: string,
	disabled?: boolean,
	svg?: string,
	price?: string,
	autoComplete?: boolean
}

//
// const upperCaseFirstLetter = (element: string) => {
// 	return element.charAt(0).toUpperCase() + element.slice(1).toLowerCase()
// }
// const removeDotFromName = (name: string) => {
// 	// return name.split('.')[0]
// }
export const RenderField = (props: IProps) => {
	const {
		key,
		input,
		label,
		type,
		disabled,
		placeholder,

		// autoComplete,
		svg,
		meta: { pristine, touched, invalid, active, dirty, error, warning }
	} = props

	const renderSvgColor: string = !pristine || touched
		? invalid
			? 'red'
			: 'green'
		: 'grey' // silver

	// console.log('error', error)
	// console.log('pristine', pristine)
	// console.log('invalid', invalid)
	// console.log('touched', touched)

	const showLabel = (): boolean => {
		if (active) {
			return true
		} else if ((dirty && !pristine) || input.value) {
			return true
		}
		return false
	}

	const messageTest = (errorObj: {} | string) => {
		if (typeof errorObj === 'string') {
			return (
				<span>{error}</span>
			)
		}
	}

	return (
		<div key={key} style={{ position: 'relative' }}>
			<input
				{...input}
				aria-label={label}
				placeholder={placeholder}
				type={type}
				disabled={disabled}
				readOnly={disabled}
			/>
			<Label className={'renderLabel'} active={showLabel()}>{label}</Label>
			{svg && <Svg className={'renderInputSvg'} color={renderSvgColor}>
				{renderSvg(svg)}
      </Svg>}

			{messageTest(error)}
			{messageTest(warning)}

		</div>
	)
}

export default RenderField

export const Svg = styled.div`
	path, rect{
		fill: ${(props: any) => props.color};
	}
	width: 25px;
`

interface ILabelProps {
	active: boolean;
	spencer?: string;
}

// Optional Styled-Components
// const Label = styled<ILabelProps, 'label'>('label')`
// 	display: block;
// 	color: ${(props) => props.active ? 'red' : 'blue'};
// `
// had to export this to test
export const Label = styled.label`
	display: block;
	color: ${(props: ILabelProps) => props.active ? 'green' : 'blue'};
`
