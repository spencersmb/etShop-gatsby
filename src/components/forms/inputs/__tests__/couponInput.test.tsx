import { CouponInput } from '@components/forms/inputs/couponInput'
import initialState from '@redux/reducers/initialState'
import { testCartWithItem, testCartWithMultiplesFixedCartCoupon } from '@redux/reduxTestUtils'
import React from 'react'
import rxjs from 'rxjs'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup,
	fireEvent,
	waitForElement
} from 'react-testing-library'
import 'jest-dom/extend-expect'
import { TestScheduler } from 'rxjs/testing'

afterEach(cleanup)

const propsDefault = {
	checkCoupon: jest.fn(),
	loadCoupon: jest.fn(),
	invalidCoupon: jest.fn(),
	submitCoupon: jest.fn(),
	updatePrice: jest.fn(),
	coupon: initialState.cart.coupon,
	total: 12,
	cartItems: initialState.cart.items
}
const propsValid = {
	checkCoupon: jest.fn(),
	loadCoupon: jest.fn(),
	invalidCoupon: jest.fn(),
	submitCoupon: jest.fn(),
	updatePrice: jest.fn(),
	coupon: {
		...testCartWithMultiplesFixedCartCoupon.coupon
	},
	total: 12,
	cartItems: testCartWithMultiplesFixedCartCoupon.items
}
const propsInvalid = {
	checkCoupon: jest.fn(),
	loadCoupon: jest.fn(),
	invalidCoupon: jest.fn(),
	submitCoupon: jest.fn(),
	updatePrice: jest.fn(),
	coupon: {
		code: '',
		discount: '',
		loading: false,
		product_ids: [],
		submitted: true,
		type: '',
		valid: false
	},
	total: 12,
	cartItems: testCartWithItem.items
}
const propsLoading = {
	checkCoupon: jest.fn(),
	loadCoupon: jest.fn(),
	invalidCoupon: jest.fn(),
	submitCoupon: jest.fn(),
	coupon: {
		code: '',
		discount: '',
		loading: true,
		product_ids: [],
		submitted: false,
		type: '',
		valid: false
	}
}
// const scheduler = new TestScheduler(0);
describe('Coupon Input', () => {

	it('renders correctly', () => {
		const tree = renderer
			.create(
				<CouponInput {...propsDefault}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should display valid class', () => {
		const modalRender = render(<CouponInput {...propsValid}/>)
		const notice = modalRender.getByTestId('formGroupTest')
		expect(notice).toHaveClass('valid')
	})

	it('Should display valid code text', () => {
		const modalRender = render(<CouponInput {...propsValid}/>)
		const input: any = modalRender.getByTestId('couponInput')
		expect(input.value).toBe(propsValid.coupon.code)
	})

	it('Should display invalid class', () => {
		const modalRender = render(<CouponInput {...propsInvalid}/>)
		const notice = modalRender.getByTestId('formGroupTest')
		expect(notice).toHaveClass('invalid')
	})

	// xit('Should not call api if input is blank', () => {
	// 	const modalRender = render(<CouponInput {...propsDefault}/>)
	// 	const btn = modalRender.getByTestId('couponSubmitBtn')
	// 	btn.click()
	// 	expect(propsDefault.checkCoupon).toHaveBeenCalledTimes(0)
	// })

	// it('Should call api when code is entered', async () => {
	// 	const scheduler = new TestScheduler(0);
	// 	const modalRender = render(<CouponInput {...propsDefault}/>)
	// 	const input = modalRender.getByTestId('couponInput')
	// 	fireEvent.input(input, { target: { value: 'test-code' } })
	// 	// expect(propsDefault.submitCoupon).toHaveBeenCalledTimes(1)
	// 	const greetingNode: any = await waitForElement(() => modalRender.getByTestId('couponInput'))
	// 	expect(greetingNode.value).toBe('test-code')
	// 	expect(propsDefault.submitCoupon).toHaveBeenCalledTimes(1)
	// })

	// xit('Should show btn text Checking...', () => {
	// 	const modalRender = render(<CouponInput {...propsLoading}/>)
	// 	const btn = modalRender.getByTestId('couponSubmitBtn')
	// 	expect(btn.innerHTML).toBe('Checking...')
	// })

})
