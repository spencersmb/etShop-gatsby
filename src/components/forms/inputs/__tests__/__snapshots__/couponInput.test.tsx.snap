// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon Input renders correctly 1`] = `
<div
  className="sc-htoDjs kZkykA"
>
  <div
    className="sc-jzJRlG gBiMFt"
  >
    COUPON
  </div>
  <form>
    <div
      className="formGroup__Container sc-eNQAEJ CbWlr"
      data-testid="formGroup"
    >
      <div
        className="formInput sc-hMqMXs jwjJTg"
      >
        <div
          className="formGroup noFocus invalid undefined"
          data-testid="formGroupTest"
        >
          <label
            className="renderLabel"
            htmlFor="couponInput"
          >
            Enter Coupon Code
          </label>
          <input
            aria-label="coupon"
            data-testid="couponInput"
            defaultValue=""
            disabled={false}
            id="couponInput"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
          />
          <div
            className="submit__spinner sc-dnqmqq iKaPWV"
            data-testid="spinner"
          >
            <svg
              className="spinner"
              viewBox="0 0 50 50"
            >
              <circle
                className="path"
                cx="25"
                cy="25"
                fill="none"
                r="20"
                strokeWidth="6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
